fedimint-api/src/db/mod.rs

```diff
@@ -48,7 +50,13 @@ pub trait Database: Send + Sync {
     fn raw_apply_batch(&self, batch: DbBatch) -> Result<()>;
 }
 
-impl<'a> dyn Database + 'a {
+dyn_newtype_define! {
+    /// A handle to a type-erased database implementation
```

Comment:
Notably the macro handles corectly both docstring and `#[derive(...)]`.

![Screenshot from 2022-09-23 12-22-37](https://user-images.githubusercontent.com/9209/192042243-c42526ec-677d-4555-87ce-44cdddca212f.png)